src/Features/LanguageServer/ProtocolUnitTests/SpellCheck/SpellCheckTests.cs

```diff
@@ -64,10 +65,47 @@ public async Task TestDocumentResultsForOpenFiles(bool mutatingLspWorkspace)
             AssertJsonEquals(results.Single(), new VSInternalSpellCheckableRangeReport
             {
                 ResultId = "DocumentSpellCheckHandler:0",
-                Ranges = GetRanges(sourceText, testDocument.AnnotatedSpans),
+                Ranges = GetRanges(testDocument.AnnotatedSpans),
             });
         }
 
+        [Theory, CombinatorialData]
+        public async Task TestLotsOfResults(bool mutatingLspWorkspace)
+        {
+            // Produce an 'interesting' large string, with varying length identifiers, and varying distances between the spans. 
+            var random = new Random(0);
+            var markup = string.Join(Environment.NewLine, Enumerable.Range(0, 5500).Select(v =>
+$$"""
+class {|Identifier:A{{v}}|}
+{
+}
+{{string.Join(Environment.NewLine, Enumerable.Repeat("", random.Next() % 5))}}
+"""));
+            await using var testLspServer = await CreateTestLspServerAsync(markup, mutatingLspWorkspace);
+
+            // Calling GetTextBuffer will effectively open the file.
+            var testDocument = testLspServer.TestWorkspace.Documents.Single();
+            testDocument.GetTextBuffer();
```

Comment:
this shouldn't be needed I don't think - a relic of old code.

Reply:
thanks.  removed!